addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -733,7 +733,7 @@ var SnippetEditor = Widget.extend({
         var $element = this.$target.parent();
         while ($element.length) {
             var parentEditor = $element.data('snippet-editor');
-            if (parentEditor) {
+            if (parentEditor && parentEditor._customize$Elements) {
```

Comment:
Ok I did a bisect, and the commit that introduced the bug is https://github.com/odoo/odoo/commit/91293fe4a8125f65cd7e5f487aacbc62c35c0f74
That's not a "little" diff :)

Reply:
Ouf, that will be difficult to find the cause looking at that commit indeed... all that's left is more debugging :grimacing:

Reply:
After investigating, I’ve retargeted the fix to 17.3. The issue with the OWL conversion is that, in some cases, when the SnippetEditor widget is instantiated, the init method isn’t called before the start method. I’m still looking into why this happens. As a result, _customize$Elements isn’t just empty; it’s undefined, which leads to an array with an undefined element at the start, causing the bug. The current solution I’ve implemented is to initialize _customize$Elements in the init method to prevent the crash.
Before the "bad" commit : 
![image](https://github.com/user-attachments/assets/3bfd8657-0a3f-465f-b783-944cd9e99795)

After the "bad" commit :
![image](https://github.com/user-attachments/assets/779ffe36-c778-4a98-88b3-793fa4398e07)

It's very strange. If you have an idea, i'm open...

Reply:
Clarification, that happens when you're click on Edit to open the Snippet Menu

Reply:
@qsm-odoo Do you think this "temporary" bug fix will do the job for the time being?

Reply:
@SergeBayet I missed your previous messages somehow.

> the init method isn’t called before the start method.

That's impossible: `init` is the method called when you instantiate something, `start` cannot be called before as it does not exist. What your screenshots show is that some editor instance `start` comes before some other editor instance `init` but how can you be sure it is the same instances? (based on those screenshots)

> Do you think this "temporary" bug fix will do the job for the time being?

Well I guess if it prevents a traceback, we can merge it with a FIXME comment alongside it but we definitely would need to investigate that in a task afterwards. Because doing this, it means some parent options do not appear in the right panel anymore, right?

Reply:
@qsm-odoo 
I identified the issue after realizing that the code (in the OWL conversion commit) uses `Promise.all` [here](https://github.com/odoo/odoo/commit/91293fe4a8125f65cd7e5f487aacbc62c35c0f74#diff-52a4f9d2c217548e69e6b7fd097f286f1754a6389734eea254b87255e501cbefR2678-R2687) which caused the elements and their descendants to be processed in parallel. This led to out-of-order execution and inconsistencies when handling nested elements. By refactoring the code to use a `for` loop with `await`, I ensured that each element (and its descendants) is processed sequentially, which resolved the bug. What do you think?

Reply:
@qsm-odoo If you have time, could you tell me if this suits you and r+ if it does? Thanks

Reply:
TBH not really :(  It seems to be a corner case. I'll dig into it next days.

Reply:
Okaaay, I now understand everything about this, that was not really easy to understand :grimacing: Thanks for the detailed commit @bvr-odoo and sorry for not checking this sooner @SergeBayet : despite being a good fix, I was pretty sure there was something more to find out and with the OXP etc, I focused on other bugs. Indeed, what you did would solve the issue, but unfortunately I think that Arthur was right to remove the sequential creation of editors: it was made so that this can be parallelized. I'll close this one then.

I'll comment on the other PR to make a few changes @bvr-odoo :pray: